### PR TITLE
Add SysWhispers2 detection & add 0x2e syscall detection

### PR DESCRIPTION
**Pull request by Still34**
## Summary

This PR adds rudimentary support for detecting SysWhisper2 syscall list population and adding the 0x2e syscall used by the project to the nursery collection.

### CODE REVIEW

File: nursery/execute-syscall-instruction.yml [hunk 1]
```diff
@@ -5,15 +5,21 @@ rule:
...
       static: basic block
       dynamic: unsupported  # requires mnemonic features
     references:
       - https://github.com/j00ru/windows-syscalls
+      - https://codemachine.com/articles/system_call_instructions.html
   features:
     - and:
-      - mnemonic: syscall
+      - or:
+        - mnemonic: syscall
```
Comment (mr-tz) [inline nursery/execute-syscall-instruction.yml:19]: should we add `sysenter` and rename the rule accordingly?

Comment (mr-tz) [reply nursery/execute-syscall-instruction.yml:19]: or create a new rule?

File: nursery/populate-syswhispers2-syscall-list.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    - and:
+      - and:
+        - description: check ntdll.dll
+        - instruction:
+          - mnemonic: or
+          - operand[1].number: 0x20202020
+        - or:
+          - instruction:
+            - mnemonic: cmp
+            - operand[1].number: 0x6C64746E
```
Comment (mr-tz) [inline nursery/populate-syswhispers2-syscall-list.yml:28]:
```suggestion
            - operand[1].number: 0x6C64746E = "ldtn"
```

File: nursery/populate-syswhispers2-syscall-list.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+          - instruction:
+            - mnemonic: cmp
+            - operand[1].number: 0x6C642E6C
```
Comment (mr-tz) [inline nursery/populate-syswhispers2-syscall-list.yml:31]:
```suggestion
            - operand[1].number: 0x6C642E6C = "ld.l"
```

File: nursery/populate-syswhispers2-syscall-list.yml [hunk 4]
```diff
@@ -0,0 +1,44 @@
+rule:
+  meta:
+    name: populate syswhispers2 syscall list
```
Comment (mr-tz) [inline nursery/populate-syswhispers2-syscall-list.yml:3]:
```suggestion
    name: populate SysWhispers2 syscall list
```

File: nursery/populate-syswhispers2-syscall-list.yml [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+      - Defense Evasion::Obfuscated Files or Information::Indicator Removal from Tools [T1027.005]
+    references:
+      - https://github.com/jthuraisamy/SysWhispers2/blob/main/data/base.c#L32
+    examples:
+      - 932dab8756ad4ae9a62bde0772d952e4:0x14001BD1E
+  features:
+    - and:
+      - and:
+        - description: check ntdll.dll
+        - instruction:
```
Comment (mr-tz) [inline nursery/populate-syswhispers2-syscall-list.yml:22]:
```suggestion
        - instruction:
          - description: "to lower case"
```

File: nursery/populate-syswhispers2-syscall-list.yml [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+            - operand[1].number: 0x6C64746E
+          - instruction:
+            - mnemonic: cmp
+            - operand[1].number: 0x6C642E6C
+      - and:
+        - description: check system call starting with 'Zw'
+        - instruction:
+          - description: check 'Zw'
+          - mnemonic: cmp
+          - operand[1].number: 0x775a
```
Comment (mr-tz) [inline nursery/populate-syswhispers2-syscall-list.yml:37]:
```suggestion
          - operand[1].number: 0x775a = "wZ"
```

**Review by mr-tz — COMMENTED** [review]
thank you, a few suggestions/comments inline

File: nursery/populate-syswhispers2-syscall-list.yml
Comment (mr-tz) [inline nursery/populate-syswhispers2-syscall-list.yml:1]: we can move this rule from the nursery

**Comment by mr-tz** [comment]
hey @Still34 can you take a look at the comments so we can hopefully merge this?

**Comment by Still34** [comment]
Hey sorry I didn't see the stray comments until now - I'll get around to it after the holiday.

File: nursery/populate-syswhispers2-syscall-list.yml
Comment (Still34) [reply nursery/populate-syswhispers2-syscall-list.yml:1]: Where would you like me to move it to?

Comment (mr-tz) [reply nursery/populate-syswhispers2-syscall-list.yml:1]: we use the namespace `linking/runtime-linking` -- do you think this one is the best fit?

**Pull request merged**